pro_tes/middleware/task_distribution/distance.py

```diff
@@ -44,15 +44,22 @@ def task_distribution(input_uri: List) -> List:
     access_uri_combination = get_uri_combination(input_uri, tes_uri)
 
     # get the combination of the tes ip and input ip
-    ips = ip_combination(input_uri=input_uri, tes_uri=tes_uri)
+    try:
+        ips = ip_combination(input_uri=input_uri, tes_uri=tes_uri)
+    except (InputUriError, TesUriError):
+        return random.task_distribution()
+
```

Comment:
I think the different middlewares should not be coupled. So I think this should be implemented _outside_ of `distance.py`, i.e., raise the error here, then catch it in the calling context and use the alternative middleware instead.